recyclerview/src/main/java/me/henrytao/recyclerview/BaseAdapter.java

```diff
@@ -126,10 +138,14 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
   }
 
   public int getBaseItemCount() {
-    if (mBaseAdapter == null || !mIsBaseAdapterEnabled) {
+    if (mBaseAdapters == null || !mIsBaseAdapterEnabled) {
       return 0;
     }
-    return mBaseAdapter.getItemCount();
+    int count = 0;
+    for (RecyclerView.Adapter adapter : mBaseAdapters) {
```

Comment:
This is simple but super slow. Check out http://howtodoinjava.com/2013/03/26/performance-comparison-of-different-for-loops-in-java/

We should use

```
int i = 0;
for (int n = mBaseAdapters.length; i < n; i++) {
    ...
}
```